security/pkg/nodeagent/caclient/providers/google/client.go

```diff
@@ -85,7 +88,16 @@ func (cl *googleCAClient) CSRSign(ctx context.Context, csrPEM []byte, token stri
 		token = bearerTokenPrefix + token
 	}
 
-	ctx = metadata.NewOutgoingContext(ctx, metadata.Pairs("Authorization", token))
+	out, _ := metadata.FromOutgoingContext(ctx)
+	out = out.Copy()
```

Comment:
Could you add a comment explaining why Copy() is needed here (preventing races by modification)?